libraries/SuplaDevice/supla-common/srpc.c

```diff
@@ -1409,7 +1409,8 @@ _supla_int_t SRPC_ICACHE_FLASH srpc_ds_async_registerdevice(
       (sizeof(TDS_SuplaDeviceChannel) * SUPLA_CHANNELMAXCOUNT) +
       (sizeof(TDS_SuplaDeviceChannel) * registerdevice->channel_count);
 
-  if (size > sizeof(TDS_SuplaRegisterDevice)) return 0;
+  if (size > sizeof(TDS_SuplaRegisterDevice)) return SUPLA_RESULT_DATA_TOO_LARGE;
```

Comment:
These functions cannot return non-zero errors. It is possible that srpc_async_call returns -4 as the request id

Reply:
Right, then only "0" can indicate problem. So current implementation also contain such bug, since it can return SUPLA_RESULT_CALL_NOT_ALLOWED which is -5

Reply:
I know that. Therefore, SUPLA_RESULT_CALL_NOT_ALLOWED will be replaced with SUPLA_RESULT_FALSE. For now, the chance that rr_id reaches -5 is small. It will only occur after 2 billion calls.